First commit

EvgenyYAgit opened the pull request:
Первый коммит. Тесты проходят успешно. Отчет Allure записан.

EvgenyYAgit committed:
First commit

irisqul commented on pages/order_scooter.py:
```diff
@@ -0,0 +1,86 @@
+        self.driver = driver
+
+    # переход на страницу
+    def get_site(self, site):
```
Нужно исправить: нужен еще родительский класс BasePage - там разместятся методы общие для всех страниц

irisqul commented on pages/order_scooter.py:
```diff
@@ -0,0 +1,86 @@
+
+    # переход на страницу
+    def get_site(self, site):
+        self.driver.get(site)
```
Можно лучше: нет смысла дублировать методы которые уже существуют на уровне драйвера. мы создаем новые объекты страниц чтобы добавить функционал которого еще нет, а не переименовать то что уже работает.

irisqul commented on pages/order_scooter.py:
```diff
@@ -0,0 +1,86 @@
+        self.driver.get(site)
+
+    # прокрутка до элемента на странице
+    def scroll_element(self, scroll_element):
```
Нужно исправить: например это метод для basePage, и некоторые далее тоже

irisqul commented on pages/order_scooter.py:
```diff
@@ -0,0 +1,86 @@
+            expected_conditions.element_to_be_clickable(element_clickable))
+
+
+class OrderPage:
```
Нужно исправить: каждый РО нужно разместить в отдельном файле

irisqul commented on pages/order_scooter.py:
```diff
@@ -0,0 +1,86 @@
+        self.driver.find_element(*string_when_bring_scooter).send_keys(Keys.ENTER)
+
+    # прокрутка до элемента
+    def scroll_element(self, scroll_element):
```
Нужно исправить: такой метод уже был, нужно наследовать его от basePage когда он появится

irisqul commented on locators/order_scooter_locators.py:
```diff
@@ -0,0 +1,31 @@
+
+site = 'https://qa-scooter.praktikum-services.ru/'
+date = "15.05.2024" #дата аренды
+comment_for_courier = "Позвоните за час до приезда" #комментарий
```
Нужно исправить: константы, тестовые данные, урлы, и локаторы - это разная информация. Нужно разделять на разные файлы и классы.  Из локаторов нужно выделить файл с урлами и путями страниц, и тестовые данные

irisqul commented on locators/list_questions_locators.py:
```diff
@@ -0,0 +1,23 @@
+from selenium.webdriver.common.by import By
+
+site = 'https://qa-scooter.praktikum-services.ru/'
```
up, урлы нужно вынести и переиспользовать

irisqul commented on tests/test_list_questions.py:
```diff
@@ -0,0 +1,114 @@
+    driver = None
+
+    @classmethod
+    def setup_class(cls):
```
Нужно исправить: давай используем пройденную архитектуру - нужно добавить фикстуру, которая запускает и завершает работу браузера

irisqul commented on tests/test_list_questions.py:
```diff
@@ -0,0 +1,114 @@
+        home_page.scroll_element(question_how_much_it_cost)
+        home_page.wait_element_clickable(question_how_much_it_cost)
+        home_page.click_on_section(question_how_much_it_cost)
+        home_page.wait_element_located(answer_how_much_it_cost)
```
Нужно исправить: объединить методы в осмысленный шаг - убрать ожидания и прочие технические действия внутрь шага по клику на вопрос

irisqul commented on tests/test_list_questions.py:
```diff
@@ -0,0 +1,114 @@
+        cls.driver = webdriver.Firefox()
+
+    @allure.title('Проверка ответа на вопрос: Сколько это стоит? И как оплатить?')
+    def test_question_how_much_it_cost(self):
```
Можно лучше: кажется тут отлично подойдет параметризация - проверить разные вопросы

irisqul commented on tests/test_list_questions.py:
```diff
@@ -0,0 +1,114 @@
+        assert answer == 'Да, обязательно. Всем самокатов! И Москве, и Московской области.'
+
+    @classmethod
+    def teardown_class(cls):
```
Нужно исправить: это переносим в фикстуру

irisqul commented on tests/test_order_scooter.py:
```diff
@@ -0,0 +1,103 @@
+from pages.order_scooter import HomePage, OrderPage
+import pytest
+import allure
+from locators.order_scooter_locators import site, button_little_order, button_middle_order, button_big_order, \
```
Нужно исправить: можно просто все импортировать, не обязательно перечислять что именно

irisqul commented on tests/test_order_scooter.py:
```diff
@@ -0,0 +1,103 @@
+
+    @classmethod
+    def setup_class(cls):
+        cls.driver = webdriver.Firefox()
```
up

irisqul commented on tests/test_order_scooter.py:
```diff
@@ -0,0 +1,103 @@
+        order_page.click_on_section(color_scooter_black)
+        order_page.filling_string(string_comment_for_courier, comment_for_courier)
+        order_page.click_on_section(button_middle_order)
+        order_page.click_on_section(button_yes)
```
Нужно исправить: пожалуйста, собери осмысленные шаги. Все техническое должно остаться в РО

irisqul commented on tests/test_order_scooter.py:
```diff
@@ -0,0 +1,103 @@
+        order_page.click_on_section(button_yes)
+
+        assert "Заказ оформлен" in order_page.get_text_of_element(windows_success_order)
+
```
Нужно исправить: тест неатомарный после ассерта шагов быть не должно

EvgenyYAgit committed:
Создан родительский класс BasePage - там разместятся методы общие для всех страниц

EvgenyYAgit committed:
Commit 3
Был создан отдельный общий класс с общим фунционалом для всех тестов, он вынесен в отдельный файл. По итогу файл с классом для тестов с вопросами был удален, из-за того что все есть в общем классе.

EvgenyYAgit committed:
Commit 4
Из локаторов нужно выделить файл с урлами и путями страниц, и тестовые данные. Вынес тестовые данные в один файл, url так как используется во всех тестах вынес тоже отдельно.

EvgenyYAgit committed:
Commit 5
Фикстура на старт и закрытие браузера написана и добавлена в тесты.

EvgenyYAgit committed:
Commit 6
Объединил методы в осмысленный шаг - убрал ожидания и прочие технические действия внутрь шага по клику на вопрос.
Пришлось все-таки создать еще один метод именно для этого объединеного шага.

EvgenyYAgit committed:
Commit 7
Чтобы не было каши импортировал сразу весь файл в каждом тесте, и использовал значения из этого файла.

EvgenyYAgit committed:
Commit 8
Собрал осмысленные шаги в тестах. Стали более компактными и лаконичными.

EvgenyYAgit committed:
Commit 9
Новые отчеты Allure по успешном прохождении тестов.

irisqul commented on tests/test_list_questions.py:
```diff
@@ -0,0 +1,86 @@
+
+    @allure.title('Проверка ответа на вопрос: Сколько это стоит? И как оплатить?')
+    def test_question_how_much_it_cost(self, start_and_stop_browser):
+        home_page = BasePage(start_and_stop_browser)
```
Нужно исправить: внутри тестов вызываются классы конкретных страниц, BasePage - это надкласс страниц, в тестах его быть не должно. Все его методы унаследованы классами отдельных страниц.

irisqul commented on pages/basepage.py:
```diff
@@ -0,0 +1,49 @@
+from selenium.webdriver.support.wait import WebDriverWait
+
+
+class BasePage:
```
Нужно исправить: BasePage - это не замена HomePage, в BasePage  будут лежать методы общие для всех страниц, а в хоумпейдж(судя по названию) - методы для главной страницы

EvgenyYAgit committed:
Commit 10
Переделал, родительский класс отдельно, от него наследует другой класс основные общие методы.
Тесты проходтя успешно, все ОК.
В процессе написания и исправления проекта, научился его писать. Спасибо!

EvgenyYAgit committed:
Commit 11
Вернул РО для home page. Создал уникальный метод для test_list_questions.py
Остальные общие методы берутся из basepage.
